packages/http-sdk/src/authz/authz-http.service.ts

```diff
@@ -0,0 +1,136 @@
+      return this.isValidFeeAllowance(response.allowance) ? response.allowance : undefined;
+    } catch (error) {
+      if (error.response.data.message.includes("fee-grant not found")) {
+        return undefined;
+      }
+
+      throw error;
+    }
+  }
+
+  async getDepositDeploymentGrantsForGrantee(address: string) {
+    const response = this.extractData(await this.get<DepositDeploymentGrantResponse>(`cosmos/authz/v1beta1/grants/grantee/${address}`));
+    return response.grants.filter(grant => this.isValidDepositDeploymentGrant(grant));
+  }
+
+  async getDepositDeploymentGrantsForGranterAndGrantee(granter: string, grantee: string): Promise<ExactDepositDeploymentGrant | undefined> {
+    const response = this.extractData(
+      await this.get<DepositDeploymentGrantResponse<ExactDepositDeploymentGrant>>("cosmos/authz/v1beta1/grants", {
+        params: {
+          grantee: grantee,
+          granter: granter
+        }
+      })
+    );
+    return response.grants.find(grant => this.isValidDepositDeploymentGrant(grant));
+  }
+
+  async hasValidFeeAllowance(granter: string, grantee: string) {
+    const feeAllowances = await this.getFeeAllowancesForGrantee(grantee);
+    return feeAllowances.some(allowance => allowance.granter === granter);
+  }
+
+  async hasValidDepositDeploymentGrant(granter: string, grantee: string) {
+    const depositDeploymentGrants = await this.getDepositDeploymentGrantsForGrantee(grantee);
+    return depositDeploymentGrants.some(allowance => allowance.granter === granter);
+  }
+
+  async paginateDepositDeploymentGrants(
+    options: ({ granter: string } | { grantee: string }) & { limit: number },
+    cb: (page: DepositDeploymentGrantResponse["grants"]) => Promise<void>
+  ) {
+    let nextPageKey: string | null = null;
+    const side = "granter" in options ? "granter" : "grantee";
+    const address = "granter" in options ? options.granter : options.grantee;
+
+    do {
+      const response = this.extractData(
+        await this.get<DepositDeploymentGrantResponse>(
+          `cosmos/authz/v1beta1/grants/${side}/${address}`,
+          nextPageKey
+            ? {
+                params: { "pagination.key": nextPageKey, "pagination.limit": options.limit }
+              }
+            : undefined
+        )
+      );
+      nextPageKey = response.pagination.next_key;
+
+      await cb(response.grants.filter(grant => this.isValidDepositDeploymentGrant(grant)));
+    } while (nextPageKey);
+  }
+
+  private isValidFeeAllowance({ allowance }: FeeAllowance) {
+    return allowance["@type"] === this.FEE_ALLOWANCE_TYPE && (!allowance.expiration || isFuture(new Date(allowance.expiration)));
+  }
+
+  private isValidDepositDeploymentGrant({ authorization, expiration }: ExactDepositDeploymentGrant) {
+    return authorization["@type"] === this.DEPOSIT_DEPLOYMENT_GRANT_TYPE && (!expiration || isFuture(new Date(expiration)));
```

Comment:
Technically undefined expiration is not supported with our custom type

Reply:
Do u think I should remove the check?

Reply:
I think we can leave it there, it should be fixed in the coming releases.